kube-derive/tests/crd_schema_test.rs

```diff
@@ -2,13 +2,14 @@
 
 use assert_json_diff::assert_json_eq;
 use chrono::{DateTime, Utc};
+use kube::ValidateSchema;
 use kube_derive::CustomResource;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 use std::collections::{HashMap, HashSet};
 
 // See `crd_derive_schema` example for how the schema generated from this struct affects defaulting and validation.
-#[derive(CustomResource, Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+#[derive(CustomResource, Serialize, Deserialize, Debug, PartialEq, Clone, ValidateSchema)]
```

Comment:
Is the idea here now is to replace `JsonSchema` with `ValidateSchema` whenever we need to use `rule` annotations? I see it's not a universal replacement, the sub-structs still use `JsonSchema`.

Reply:
It is a wrapper on top of the `JsonSchema` implementing it. Crate path for the JsonSchema can be overridden, same as in the CustomResource trait. Rule is one of the keywords which can be used here, there maybe more in the future, since the implementation is extensible and will not be affected by `schemars` versions. Sub-schemas don’t need `ValidateSchema`, if a rule is not used there. Implementing this would require to blanket implement the `ValidateSchema` for all standard types, and since we only modify <this current schema I see now> I tried to keep it simple, and it also was the only path forward.

Reply:
I moved towards implementing option 3 from [comment](https://github.com/kube-rs/kube/issues/1367#issuecomment-2495979780), so current changes are direct descendants of this (and the fact that I realized how to work around unordered properties in the schema)